pkg/store/proxy.go

```diff
@@ -104,15 +123,69 @@ func (s *ProxyStore) Info(ctx context.Context, r *storepb.InfoRequest) (*storepb
 	res.MaxTime = MaxTime
 	res.MinTime = MinTime
 
-	for _, l := range s.selectorLabels {
-		res.Labels = append(res.Labels, storepb.Label{
-			Name:  l.Name,
-			Value: l.Value,
+	if len(stores) == 0 {
+		return res, nil
+	}
+
+	g, gctx := errgroup.WithContext(ctx)
+	respSender, respRecv, closeFn := newInfoRespCh(gctx, 10)
```

Comment:
I'm a bit worried about this part, whether it will actually work with all the time outs and partial response strategy.

So IMO more tests is needed.

Also maybe this could be simplified using mutex? Channels are a bit hard to deal with :/

Reply:
I don't think there is a possibility for tolerating partial responses. We essentially need a consistent system for metadata like this, meaning either all succeed or none, and metadata stays unchanged until the next successful request.

Adding more tests is definitely a good idea, on that.

Reply:
Hm.. Why we are doing this here? (: 

Querier already maintain healthy nodes in storeset. It does it by periodic `Info` calls. This always include labels as well. I think we can rely on this asyncronous Info calls here as well instead of duplicated, sync forwarding. What do you think? (:

I think some lag is fine with benefits of reduced number of requests + simplified code here.

Reply:
That's fair, will adapt.

Reply:
Removed all the async code here and just assembled the label-sets with the cached store's label-sets.